client/client_update_test.go

```diff
@@ -893,6 +965,53 @@ func TestUpdateNonRootRemoteCorruptedNoLocalCache(t *testing.T) {
 				role: role,
 			}, testData, true)
 		}
+
+		switch role {
+		case data.CanonicalSnapshotRole:
+			for _, missing := range []string{data.CanonicalRootRole, data.CanonicalTargetsRole} {
+				testData := swizzleExpectations{
+					desc: fmt.Sprintf("snapshot missing root meta checksum"),
+					// TODO: this should probably be always client.ErrMissingMeta
```

Comment:
wow, thanks for tracking this down.  We should add these additional checks so that this errors much more gracefully.  Either in this PR or another one to keep it more manageable (and I can help implement)